packages/cozy-mespapiers-lib/src/helpers/queries.js

```diff
@@ -50,7 +51,8 @@ export const buildFilesQueryWithQualificationLabel = () => {
         })
         .partialIndex({
           'metadata.qualification.label': {
-            $exists: true
+            $exists: true,
+            ...(flag('hide.healthTheme.enabled') && { $nin: healthLabels })
```

Comment:
I think this is dangerous: your partial index MUST be static, i.e. the values should not change. Otherwise, you will create a new index each time your `healthLabels`change.

Reply:
nouvelle approche : on ne touche plus aux requêtes